tests/cli/test_run.py

```diff
@@ -246,3 +246,124 @@ def test_main_cannot_write_zuliprc_given_good_credentials(
+        "footlinks": "disabled",
+        "notify": "enabled",
+        "color-depth": "256"
+    },
+    {
+        "theme": "default",
+        "autohide": "autohide",
+        "maximum-footlinks": "3",
+        "notify": "enabled",
+        "color-depth": "256"
+    }])
+def test_successful_main_function_with_config(
+        parameterized_zuliprc, zulip_config, minimal_zuliprc,
+        capsys, mocker
+                                             ):
+    modified_zuliprc = parameterized_zuliprc(minimal_zuliprc, zulip_config)
+    mocker.patch('zulipterminal.core.Controller.__init__',
+                 side_effect=SystemExit)
+    with pytest.raises(SystemExit):
+        main(['-c', modified_zuliprc])
+    if "footlinks" in zulip_config:
+        footlinks_val = "   footlinks setting 'disabled'"
+    else:
+        footlinks_val = "   maximum footlinks setting '3'"
+    footlinks_output = "{footlinks_val} specified in zuliprc file.".format(
+        footlinks_val=footlinks_val
+    )
+    captured = capsys.readouterr()
+    lines = captured.out.strip().split("\n")
+    expected_lines = [
+        'Loading with:',
+        "   theme 'zt_dark' specified in zuliprc file (by alias 'default').",
+        "   autohide setting 'autohide' specified in zuliprc file.",
+        footlinks_output,
+        "   color depth setting '256' specified in zuliprc file."
+                     ]
+    assert lines == expected_lines
+
+
+@pytest.mark.parametrize('zulip_config', [
+    {
+        "footlinks": "enabled",
+        "maximum-footlinks": "3"
+    }])
+def test_successful_main_function_with_footlinks_and_maximum_footlinks(
+        parameterized_zuliprc, zulip_config, minimal_zuliprc, capsys,
+        mocker
+                                                                      ):
+    modified_zuliprc = parameterized_zuliprc(minimal_zuliprc, zulip_config)
+    mocker.patch('zulipterminal.core.Controller.__init__',
+                 side_effect=SystemExit)
+    with pytest.raises(SystemExit) as e:
+        main(['-c', modified_zuliprc])
+        assert str(e.value) == "1"
+    captured = capsys.readouterr()
+    lines = captured.out.strip()
+    expected_lines = "Both formats are not allowed"
+    assert lines == expected_lines
+
+
+@pytest.mark.parametrize('zulip_config', [
+    {
+        "maximum-footlinks": "5"
+    }])
+def test_successful_main_function_with_bad_maximum_footlinks(
+        parameterized_zuliprc, zulip_config, minimal_zuliprc,
+        capsys, mocker
+                                                            ):
+    modified_zuliprc = parameterized_zuliprc(minimal_zuliprc, zulip_config)
+    mocker.patch('zulipterminal.core.Controller.__init__',
+                 side_effect=SystemExit)
+    with pytest.raises(SystemExit) as e:
+        main(['-c', modified_zuliprc])
+        assert str(e.value) == "1"
+    captured = capsys.readouterr()
+    lines = captured.out.strip()
+    expected_lines = "Maximum value of 3 is allowed"
```

Comment:
This wasn't the logic you had before? The maximum (threshold) is intended to be the maximum number of footlinks shown with message content in the message list, and I'm not convinced we need to constrain users to a particular number? Less than 0 makes little sense, though we could treat negative as unlimited.

Reply:
I inherited this part of the logic from the previous pr. That was reviewed, so I thought this was a limiting scenario.